src/condor_starter.V6.1/execute_dir_monitor.h

```diff
@@ -0,0 +1,72 @@
+/***************************************************************
+ *
+ * Copyright (C) 1990-2007, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+#ifndef _EXECUTE_DIR_MONITOR_H
+#define _EXECUTE_DIR_MONITOR_H
+
+#include "condor_sys_types.h"
+#include <string>
+#include <tuple>
+
+/*
+*	Base Abstract Class to monitor the jobs working directory:
+*	    - Disk Usage
+*/
+class ExecDirMonitor {
+public:
+	ExecDirMonitor() = default;
```

Comment:
If we just need the default ctor/dtor/copy  ctor/move ctor, might want to go with "rule of zero" (with a comment saying  you are intentionally rule-of-zero'ing): 

[https://en.cppreference.com/w/cpp/language/rule_of_three](https://en.cppreference.com/w/cpp/language/rule_of_three
)